packages/core/src/modules/proofs/protocol/v2/V2ProofService.ts

```diff
@@ -0,0 +1,808 @@
+import { InjectionSymbols } from '../../../../constants'
+import { AriesFrameworkError } from '../../../../error'
+import { DidCommMessageRepository, DidCommMessageRole } from '../../../../storage'
+import { Wallet } from '../../../../wallet/Wallet'
+import { AckStatus } from '../../../common'
+import { ConnectionService } from '../../../connections'
+import { ProofEventTypes } from '../../ProofEvents'
+import { ProofService } from '../../ProofService'
+import { ProofsUtils } from '../../ProofsUtil'
+import { PresentationProblemReportReason } from '../../errors/PresentationProblemReportReason'
+import { V2_INDY_PRESENTATION_PROPOSAL, V2_INDY_PRESENTATION_REQUEST } from '../../formats/ProofFormats'
+import { IndyProofFormatService } from '../../formats/indy/IndyProofFormatService'
+import { ProofProtocolVersion } from '../../models/ProofProtocolVersion'
+import { ProofState } from '../../models/ProofState'
+import { PresentationRecordType, ProofRecord, ProofRepository } from '../../repository'
+
+import { V2PresentationProblemReportError } from './errors'
+import { V2PresentationAckHandler } from './handlers/V2PresentationAckHandler'
+import { V2PresentationHandler } from './handlers/V2PresentationHandler'
+import { V2PresentationProblemReportHandler } from './handlers/V2PresentationProblemReportHandler'
+import { V2ProposePresentationHandler } from './handlers/V2ProposePresentationHandler'
+import { V2RequestPresentationHandler } from './handlers/V2RequestPresentationHandler'
+import { V2PresentationAckMessage } from './messages'
+import { V2PresentationMessage } from './messages/V2PresentationMessage'
+import { V2PresentationProblemReportMessage } from './messages/V2PresentationProblemReportMessage'
+import { V2ProposalPresentationMessage } from './messages/V2ProposalPresentationMessage'
+import { V2RequestPresentationMessage } from './messages/V2RequestPresentationMessage'
+
+@scoped(Lifecycle.ContainerScoped)
+export class V2ProofService extends ProofService {
+  public async createProofRequestFromProposal(options: ProofRequestFromProposalOptions): Promise<ProofRequestFormats> {
+    const proofRecordId = options.proofRecord.id
+    const proposalMessage = await this.didCommMessageRepository.findAgentMessage({
+      associatedRecordId: proofRecordId,
+      messageClass: V2ProposalPresentationMessage,
+    })
+
+    if (!proposalMessage) {
+      throw new AriesFrameworkError(`Proof record with id ${proofRecordId} is missing required presentation proposal`)
+    }
+
+    let result = {}
+    for (const key of proposalMessage.formats) {
+      if (key.format === V2_INDY_PRESENTATION_PROPOSAL) {
+        for (const attachment of proposalMessage.proposalsAttach) {
+          const proofRequestJson = attachment.getDataAsJson<ProofRequest>() ?? null
+          result = {
+            indy: proofRequestJson,
+          }
+        }
+      } else {
+        // PK-TODO create Presentation Exchange request format
+      }
+    }
```

Comment:
We're going to extract support for indy and pex from the core of the framework, so we can't have any format specific code here

Reply:
Can use below section for this

```

    for (const key of Object.keys(proposalMessage.formats)) {
      const service = this.formatServiceMap[key]
      if (key === PresentationRecordType.Indy) {
        for (const attachment of proposalMessage.proposalsAttach) {
          const proofRequestJson = attachment.getDataAsJson<ProofRequest>() ?? null
          result = {
            indy: proofRequestJson,
          }
        }
      } else {
        if (service) {
          const createPresentationFormatsOptions: CreatePresentationFormatsOptions = {
            presentationAttachment: proposalMessage.proposalsAttach[0],
          }

          result = {
            ...result,
            ...(await service.createProofRequestFromProposal(createPresentationFormatsOptions)),
          }
        }
      }
    }

```

Reply:
No I meant we shouldn't have any references to indy in the v2 service.  We're working to make the framework more modular and this is an important aspect of this.